properties/base.py

```diff
@@ -135,51 +169,42 @@ def __new__(mcs, name, bases, classdict):                                  #pyli
 class HasProperties(with_metaclass(PropertyMetaclass, object)):
     """HasProperties class with properties"""
 
-    _backend_name = "dict"
-    _backend_class = dict
-    _defaults = None
+    _defaults = dict()
     _REGISTRY = dict()
 
     def __init__(self, **kwargs):
-        self._backend = self._backend_class()
-
-        # add the default listeners
+        self._backend = dict()
         self._listeners = dict()
+
+        # Register the listeners
         for _, val in iteritems(self._prop_observers):
             handlers._set_listener(self, val)
 
-        # set the defaults
-        defaults = self._defaults or dict()
-        for key, value in iteritems(defaults):
-            if key not in self._props.keys():
-                raise AttributeError(
-                    "Default input '{:s}' is not a known property".format(key)
-                )
-            if callable(value):
-                setattr(self, key, value())
-            else:
-                setattr(self, key, value)
-
-        # set the keywords
-        for key in kwargs:
+        # Set the GettableProperties from defaults - these are only set here
+        for key in self._props:
```

Comment:
You use `self._props[key]` 3 times in this loop, maybe it's a better idea to just use `iteritems` here.